src/contexts/BuySell/BuySellProvider.tsx

```diff
@@ -68,105 +59,73 @@ const BuySellProvider: React.FC = ({ children }) => {
 
-      setUniswapData(uniswapData)
+    const isExactInputTrade = !isUserBuying || activeField === 'currency'
 
-      // Populate the inactive field with API response
-      if (isUserBuying) {
-        if (activeField === 'currency') {
-          setTokenQuantity(uniswapData.display?.to_quantity)
-        } else {
-          setCurrencyQuantity(uniswapData.display?.from_quantity)
-        }
-      } else {
-        setCurrencyQuantity(uniswapData.display?.to_quantity)
-      }
+    getZeroExTradeData(
+      isExactInputTrade,
+      isUserBuying,
+      selectedCurrency.label || '',
+      buySellToken || '',
+      buySellQuantity || ''
+    ).then((data) => {
+      setZeroExTradeData(data)
+      setIsFetchingOrderData(false)
     })
   }, [
     isUserBuying,
     selectedCurrency,
     activeField,
-    targetTradeQuantity,
     buySellToken,
+    buySellQuantity,
   ])
 
   const onExecuteBuySell = useCallback(async () => {
-    if (!account || !uniswapData?.amount_in || !selectedCurrency) return
+    if (!account || !zeroExTradeData?.sellAmount || !selectedCurrency) return
 
-    let requiredBalance = new BigNumber(uniswapData?.amount_in).dividedBy(
+    let requiredBalance = new BigNumber(zeroExTradeData?.sellAmount).dividedBy(
       new BigNumber(10).pow(18)
     )
 
-    if (selectedCurrency.id === 'usdc') {
-      requiredBalance = new BigNumber(uniswapData?.amount_in || 0).dividedBy(
-        new BigNumber(10).pow(6)
-      )
+    if (selectedCurrency === 'usdc') {
+      requiredBalance = new BigNumber(
+        zeroExTradeData?.sellAmount || 0
+      ).dividedBy(new BigNumber(10).pow(6))
     }
 
     if (spendingTokenBalance?.isLessThan(requiredBalance)) return
 
-    const uniswapTradeType = getUniswapTradeType(
-      isUserBuying,
-      selectedCurrency.id,
-      uniswapData
-    )
-    const uniswapCallData = getUniswapCallData(
-      uniswapTradeType,
-      uniswapData,
-      account
-    )
-    const transactionOptions = getUniswapTransactionOptions(
-      uniswapTradeType,
-      uniswapData,
-      account
-    )
-
-    if (!uniswapCallData || !transactionOptions) return
-
-    const isSushiswapTrade =
-      buySellToken === Bitcoin2xFlexibleLeverageIndex.tokensetsId
+    const web3 = new Web3(ethereum)
 
-    const uniswapTradeTransaction = getUniswapTradeTransaction(
-      ethereum,
-      uniswapTradeType,
-      uniswapCallData,
-      transactionOptions,
-      isSushiswapTrade
-    )
+    zeroExTradeData.from = account
+    zeroExTradeData.gas = undefined // use metamask estimated gas limit
+    const tx = web3.eth.sendTransaction(zeroExTradeData)
 
     try {
       onSetTransactionStatus(TransactionStatusType.IS_APPROVING)
 
-      const transactionId = await uniswapTradeTransaction()
+      const transactionId: string = await new Promise((resolve, reject) => {
+        tx.on('transactionHash', (txId: string) => {
+          if (!txId) reject()
+          resolve(txId)
+        }).once('error', () => {
```

Comment:
What's this do?

Reply:
This is part of the logic for getting to the transaction failed screen. It justs sets the promise result to "0" if a tx fails, which is then checked for and thrown when it does fail (being caught at the bottom of the try/catch that this is in

Reply:
In this case can we just `reject()` like on 121? would also remove the need for line 128